validations/src/validations.rs

```diff
@@ -1906,7 +1906,10 @@ pub fn calculate_randpoe_threshold(
 ) -> (Hash, f64) {
     let max = u64::max_value();
     let initial_difficulty = std::cmp::max(1, initial_difficulty);
-    let target = if block_epoch <= epochs_with_initial_difficulty {
+    let target = if block_epoch <= epochs_with_initial_difficulty
+        || (after_second_hard_fork(block_epoch, get_environment())
+            && total_identities < initial_difficulty)
+    {
         max / u64::from(initial_difficulty)
     } else if total_identities == 0 || replication_factor >= total_identities {
```

Comment:
Maybe the second is not needed! :)

Reply:
Double-check what happens for `testnet` environments

Reply:
> Maybe the second branch is not needed! :)

After second hard fork is not needed but it is needed for the previous blocks

> Double-check what happens for `testnet` environments

In case of testnet, second_hard_fork returns false, so this branch does not exist